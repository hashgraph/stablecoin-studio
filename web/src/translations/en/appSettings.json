{
	"title": "Configure custom RPC/MIRROR",
	"subtitle": "Choose an option",
	"name": "Name",
	"rpc": "RPC",
	"mirrorNode": "Mirror node",
	"url": "Url",
	"apiKey": "Api Key",
	"addMirror": "Add new service",
	"addRPC": "Add RPC",
	"removeMirror": "Remove Mirror Node",
	"removeRpc": "Remove RPC",
	"header": "Header",
	"mirrorNameExists": "Mirror name already exists, if you want to update it remove it first and recreate it",
	"rpcNameExists": "RPC name already exists, if you want to update it remove it first and recreate it"
}
