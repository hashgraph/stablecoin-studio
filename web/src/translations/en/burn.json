{
    "title": "Burn tokens operation",
    "operationTitle": "Operation details",
    "amountLabel": "Enter the amount",
    "amountPlaceholder": "0",
    "originAccountPlaceholder": "0.0.0",
    "modalSuccessDesc": "{{amount}} tokens have been burned.",
    "modalAction": {
        "subtitle": "Operation details review",
        "originAccount": "Account",
        "amount": "Amount",
        "accept": "Burn tokens"
    }
}