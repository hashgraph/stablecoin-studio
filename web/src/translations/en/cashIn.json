{
    "title": "Cash in operation",
    "operationTitle": "Operation details",
    "amountLabel": "Introduce amount",
    "amountPlaceholder": "0",
    "destinationAccountLabel": "Destination account",
    "destinationAccountPlaceholder": "0.0.0",
    "modalSuccessDesc": "The cash in operation finished successfully.",
    "modalAction": {
        "subtitle": "Operation details review",
        "destinationAccount": "Destination account",
        "amount": "Amount",
        "accept": "Cash in tokens"
    }
}