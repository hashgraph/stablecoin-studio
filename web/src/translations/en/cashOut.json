{
    "title": "Cash out operation",
    "operationTitle": "Operation details",
    "amountLabel": "How much tokens?",
    "amountPlaceholder": "Introduce amount",
    "originAccountLabel": "From which account tokens will be withdrawn?",
    "originAccountPlaceholder": "Introduce account",
    "modalSuccessDesc": "The cash out operation finished successfully.",
    "modalAction": {
        "subtitle": "Operation details review",
        "originAccount": "Account to cash out",
        "amount": "Amount",
        "accept": "Make cash out"
    }
}