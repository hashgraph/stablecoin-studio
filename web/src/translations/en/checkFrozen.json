{
	"title": "Check account frozen status",
	"operationTitle": "Operation details",
	"accountLabel": "Enter the account",
	"accountPlaceholder": "0.0.0",
	"modalIsFrozen": "The account {{ account }} is frozen for the token.",
	"modalIsNotFrozen": "The account {{ account }} is unfrozen for the token.",
	"modalAction": {
		"subtitle": "Operation details review",
		"account": "Account",
		"accept": "Get account frozen status"
	}
}
