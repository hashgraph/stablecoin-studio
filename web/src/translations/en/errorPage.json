{
  "stableCoinNotSelected": {
    "title": "Choose a stable coin to start operating!",
    "description": "You have not chosen a stable coin yet.",
    "description2": "Choose an item to operate."
  }
}