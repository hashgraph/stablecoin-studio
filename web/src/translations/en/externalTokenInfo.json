{
    "common": {
      "nextStep" : "Next step",
      "goBack" : "Go back",
      "createStableCoin": "Create stable coin",
      "createNewStableCoin": "Import token",
      "cancel": "Cancel",
      "addExternalToken": "Import token"
    },
    "tabs": {
      "externalTokenInfo": "Token info"
    },
    "externalTokenInfo": {
      "stableCoinId": "Stable coin id",
      "stableCoinIdPlaceholder": "0.0.0",
      "title": "Imported token",
      "autoCheckRoles": "Do you want to get the roles automatically? This operation will incur a cost",
      "rolesPlaceholder": "Select the roles"
    },
    "optionalDetails": {
      "decimals": "Decimals",
      "expirationDate": "Expiration date",
      "finite": "Finite",
      "initialSupply": "Initial supply",
      "infinite": "Infinite",
      "placeholder": "Enter the {{placeholder}}",
      "maxSupply": "Max supply",
      "title": "Stable coin optional details",
      "typeSupply": "Type supply",
      "typeSupplyPlaceholder": "Choose type supply"
    },
    "managementPermissions": {
      "currentUserKey": "Current user key",
      "description": "Allow all the following features to be managed through smart contracts",
      "pause": "Pause",
      "freeze": "Freeze",
      "introduce": "Enter the {{name}}",
      "none": "None",
      "otherKey": "Other key",
      "placeholder": "Enter the {{placeholder}}",
      "rescue": "Rescue",
      "admin": "Admin",
      "supply": "Supply",
      "theSmartContract": "The smart contract",
      "title": "Stable coin management permissions",
      "totalSupply": "Total supply",
      "treasuryAccountAddress": "Treasury account address",
      "wipe": "Wipe"
    },
    "review": {
      "basicDetails": "Basic details",
      "managementPermissions": "Management permissions",
      "optionalDetails": "Optional details",
      "title": "Stable coin review"
    },
    "notification": {
      "title": "{{ result }} creating the stable coin",
      "descriptionSuccess": "The stable coin was created",
      "descriptionError": "Sorry, there was an error creating the stable coin"
    }
  }