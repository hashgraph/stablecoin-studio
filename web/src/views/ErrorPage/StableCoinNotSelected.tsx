import { Flex, Image, Text } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import SAFE_BOX from '../../assets/svg/safe-box.svg';

const StableCoinNotSelected = () => {
	const { t } = useTranslation('errorPage');

	return (
		<Flex
			data-testid='stable-coin-not-selected-container'
			bgColor='white'
			w='100%'
			h='100%'
			justifyContent='center'
			alignItems='center'
			flexDirection='column'
		>
			<Image
				data-testid='stable-coin-not-selected-logo'
				src={SAFE_BOX}
				alt='Safe box'
				w='140px'
				h='140px'
				mb='40px'
			/>
			<Text
				data-testid='stable-coin-not-selected-title'
				fontSize='22px'
				fontWeight='700'
				lineHeight='16px'
				mb='16px'
			>
				{t('stableCoinNotSelected.title')}
			</Text>
			<Text
				data-testid='stable-coin-not-selected-description'
				fontSize='16px'
				fontWeight='500'
				lineHeight='16px'
				maxW='500px'
				textAlign='center'
				mb='2px'
			>
				{t('stableCoinNotSelected.description')}
			</Text>
			<Text fontSize='16px' fontWeight='500' lineHeight='16px' maxW='500px' textAlign='center'>
				{t('stableCoinNotSelected.description2')}
			</Text>
		</Flex>
	);
};

export default StableCoinNotSelected;
