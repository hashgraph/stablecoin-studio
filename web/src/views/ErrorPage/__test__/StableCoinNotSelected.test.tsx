import { render } from '../../../test/index';
import StableCoinNotSelected from '../StableCoinNotSelected';
import en from '../../../translations/en/errorPage.json';

const SAFE_BOX = 'safe-box.svg';
const translations = en.stableCoinNotSelected;

describe(`<${StableCoinNotSelected.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<StableCoinNotSelected />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has logo, title and description', () => {
		const component = render(<StableCoinNotSelected />);

		const logo = component.getByTestId('stable-coin-not-selected-logo');
		const title = component.getByTestId('stable-coin-not-selected-title');
		const description = component.getByTestId('stable-coin-not-selected-description');

		expect(logo).toHaveAttribute('src', SAFE_BOX);
		expect(title).toHaveTextContent(translations.title);
		expect(description).toHaveTextContent(translations.description);
	});
});
