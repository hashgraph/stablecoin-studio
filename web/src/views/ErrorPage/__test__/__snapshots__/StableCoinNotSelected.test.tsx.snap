// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StableCoinNotSelected /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-mytqyt"
    data-testid="stable-coin-not-selected-container"
  >
    <img
      alt="Safe box"
      class="chakra-image css-1faqnpj"
      data-testid="stable-coin-not-selected-logo"
      src="safe-box.svg"
    />
    <p
      class="chakra-text css-1hg9qrt"
      data-testid="stable-coin-not-selected-title"
    >
      Choose a Stable coin to start operating!
    </p>
    <p
      class="chakra-text css-18wysbb"
      data-testid="stable-coin-not-selected-description"
    >
      You have not chosen a Stable coin yet.
    </p>
    <p
      class="chakra-text css-1vi71u4"
    >
      Choose one in the search bar above to operate.
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
